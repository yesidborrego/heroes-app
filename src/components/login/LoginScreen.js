import React, { useContext } from 'react'
import { AuthContext } from '../../auth/AuthContext';
import { types } from '../../types/types';

export const LoginScreen = ({ history }) => {
  const { dispatch } = useContext(AuthContext);
  const handleClickLogin = () => {
    dispatch({
      type: types.login,
      payload: {
        name: 'Isabella Borrego'
      }
    });
    const lastPatch = localStorage.getItem('lastPatch') || '/';
    history.replace(lastPatch);
  };

  return (
    <div className="container mt-5">
      <h1>Login Screen</h1>
      <hr/>
      <button
        className="btn btn-primary"
        onClick={handleClickLogin}
      >
        Login
      </button>
    </div>
  )
}
