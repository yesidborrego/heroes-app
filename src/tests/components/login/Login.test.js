import '@testing-library/jest-dom';
import { mount } from 'enzyme';
import { AuthContext } from '../../../auth/AuthContext';
import { LoginScreen } from '../../../components/login/LoginScreen';
import { types } from '../../../types/types';


describe('Probar el componente <LoginScreen/>', () => {
  const historyMock = {
    replace: jest.fn()
  };

  const contextValue = {
    dispatch: jest.fn(),
  }

  const wrapper = mount(
    <AuthContext.Provider value={contextValue}>
      <LoginScreen history={historyMock} />
    </AuthContext.Provider>
  )

  test('Debe mostrar correctamente el componente', () => {
    expect(wrapper).toMatchSnapshot();
  })

  test('Debe realizar el dispatch y la navegación', () => {
    const handleClickLogin = wrapper.find('button').prop('onClick');
    handleClickLogin(); // Se ejecuta la función del botón
    expect(contextValue.dispatch).toHaveBeenCalledWith({
      type: types.login,
      payload: {
        name: 'Isabella Borrego'
      }
    });
    expect(historyMock.replace).toHaveBeenCalledWith('/');

    localStorage.setItem('lastPatch', '/dc');
    handleClickLogin();
    expect(historyMock.replace).toHaveBeenCalledWith('/dc');

  })

})
