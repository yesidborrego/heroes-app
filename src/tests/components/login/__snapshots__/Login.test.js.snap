// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar el componente <LoginScreen/> Debe mostrar correctamente el componente 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login Screen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
